### PR TITLE
Fix javadoc description and package groups

### PR DESCRIPTION
**Pull request by srnagar**
- Update the description of all `package-info.java` to be consistent. 
- Add missing package-info.java and update javadoc groups for `Identity` module.

### CODE REVIEW

File: pom.client.xml [hunk 1]
```diff
@@ -369,6 +370,10 @@
                 <title>Azure Key Vault</title>
                 <packages>com.azure.keyvault*</packages>
               </group>
+              <group>
+                <title>Azure Identity</title>
+                <packages>com.azure.identity*</packages>
+              </group>
```
Comment (JonathanGiles) [inline pom.client.xml]: Sort these alphabetically - Identity before Key Vault

Comment (srnagar) [reply pom.client.xml]: Should they be sorted based on title or package? If it's based on package, event hub will move to the bottom and if it's the title, app config will move to the top. This decides the order in which the javadocs are actually generated. So, we can say, Core will be at the top and then all client library titles are alphabetically sorted?

Comment (JonathanGiles) [reply pom.client.xml]: Correct.

Comment (srnagar) [reply pom.client.xml]: Done

**Pull request merged**